server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1199,6 +1200,13 @@ protected Node(
                 remoteClusterStateService
             );
 
+            final HotToWarmTieringService hotToWarmTieringService = new HotToWarmTieringService(
+                settings,
+                clusterService,
+                clusterModule.getIndexNameExpressionResolver(),
+                clusterModule.getAllocationService()
+            );
```

Comment:
nit: put this behind feature flag too

Reply:
theres a feature flag here https://github.com/opensearch-project/OpenSearch/pull/14891/files#diff-f839350640f8bd27f602b3249215252bf07afef5d7e480b23271788af6cda331R1407